test/bluecatapi_test.rb

```diff
@@ -3,116 +3,31 @@
-      'mask_bits' => '24',
-      'name' => 'john.doe.local',
-      'network' => '10.75.1.0'
-    }]
-    @csv_string_ips = "ip_address\n10.75.1.1\n10.75.1.2\n10.75.1.3"
-    @ip_suggest_output = '{"ip":"10.75.1.20"}'
-  end
-
-  def test_csv_to_array
-    assert_equal @array_subnet, @connection.csv_to_array(@csv_string_subnet)
-  end
-
-  def test_rest_get
-    endpoint = 'test'
-    qs = 'qs'
-    req = stub_request(:get, format('%s://%s/api/1.0/%s/qs', @connection.scheme, @connection.host, endpoint)).to_return(status: 200, body: [{ 'test' => 'test' }].to_json)
-    assert_equal [{ 'test' => 'test' }].to_json, @connection.rest_get(endpoint, qs)
-    assert_requested req
-  end
-
-  def test_rest_post
-    endpoint = 'test'
-    req = stub_request(:post, format('%s://%s/api/1.0/%s/', @connection.scheme, @connection.host, endpoint)).to_return(status: 200, body: [{ 'test' => 'test' }].to_json)
-    assert_equal [{ 'test' => 'test' }].to_json, @connection.rest_post(endpoint, {})
-    assert_requested req
-  end
-
-  def test_rest_delete
-    endpoint = 'test'
-    id = 1
-    req = stub_request(:delete, format('%s://%s/api/1.0/%s/%s/', @connection.scheme, @connection.host, endpoint, id)).to_return(status: 200, body: [{ 'test' => 'test' }].to_json)
-    assert_equal [{ 'test' => 'test' }].to_json, @connection.rest_delete(endpoint, id)
-    assert_requested req
-  end
-
-  def test_doql
-    req = stub_request(:post, format('%s://%s/services/data/v1.0/query/', @connection.scheme, @connection.host)).to_return(status: 200, body: [{ 'test' => 'test' }].to_json)
-    assert_equal [{ 'test' => 'test' }].to_json, @connection.doql('test')
-    assert_requested req
-  end
-
-  def test_get_hosts
-    req = stub_request(:post, format('%s://%s/services/data/v1.0/query/', @connection.scheme, @connection.host)).to_return(status: 200, body: @csv_string_host)
-    assert_equal @array_host, @connection.get_hosts('10.75.1.1')
-    assert_requested req
-  end
-
-  def test_get_hosts_by_ip
-    req = stub_request(:post, format('%s://%s/services/data/v1.0/query/', @connection.scheme, @connection.host)).to_return(status: 200, body: @csv_string_host)
-    assert_equal @array_host, @connection.get_hosts_by_ip('10.75.1.1')
-    assert_requested req
-  end
-
-  def test_get_host_by_mac
-    req = stub_request(:post, format('%s://%s/services/data/v1.0/query/', @connection.scheme, @connection.host)).to_return(status: 200, body: @csv_string_host)
-    assert_equal @array_host, @connection.get_host_by_mac('a2:cd:f8:e9:90:eb')
-    assert_requested req
-  end
-
-  def test_get_host_by_mac_empty
-    req = stub_request(:post, format('%s://%s/services/data/v1.0/query/', @connection.scheme, @connection.host)).to_return(status: 200, body: '[]')
-    assert_equal [], @connection.get_host_by_mac('a2:cd:f8:e9:90:eb')
-    assert_requested req
+    @connection = BlueCat.new('https', true, 'bam.example.com', 123456, 'default', 123456, 'default', 123456, 'admin', 'admin')
```

Comment:
It would be great to use keyword arguments here.
In addition, the class should be namespaced, e.g. `Proxy::DHCP::BlueCat::Client`?

Reply:
will do